pkg/codegen/docs/gen.go

```diff
@@ -611,14 +611,10 @@ func (mod *modContext) typeString(t schema.Type, lang string, characteristics pr
 		displayName = mod.cleanTypeString(t, langTypeString, lang, modName, characteristics.input)
 	}
 
-	// If word-breaks need to be inserted, then the type string
-	// should be html-encoded first if the language is C# in order
-	// to avoid confusing the Hugo rendering where the word-break
-	// tags are inserted.
+	// If word-breaks need to be inserted, then the type string should be html-encoded first in order to avoid confusing
+	// the Hugo rendering where the word-break tags are inserted. This is required for languages with template types.
 	if insertWordBreaks {
-		if lang == "csharp" {
-			displayName = html.EscapeString(displayName)
-		}
+		displayName = html.EscapeString(displayName)
```

Comment:
should we just move this to the end of the function to ensure that the output is escaped and not only if there's a word break?

Reply:
#9693 has an example

Reply:
Left a comment on #9693 - I don't think that is quite right. I did update this a bit though.

Reply:
You're right. Could we move the escaping before we get into the if statement so that we can reduce duplication and adding an extra branch?

Reply:
doh ofcourse. Not enough ☕ .